From 5c57477508c86abc58cc77852b790b53e93f7440 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Thu, 4 Apr 2024 15:00:47 +0200
Subject: [PATCH 1103/1141] arm64: dts: st: use power-domain property in GPU
 node in stm32mp255.dtsi

Set SCMI GPU power domain for GPU node in stm32mp255 device tree.

Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I775372d0995c2a19a5486219bd15df234cedf9a1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/371506
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -3,6 +3,7 @@
  * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
+#include <dt-bindings/power/stm32mp25-power-domains.h>
 #include "stm32mp253.dtsi"
 
 / {
@@ -65,7 +66,7 @@
 				resets = <&rcc GPU_R>;
 				clock-names = "axi", "core";
 				clocks = <&rcc CK_BUS_GPU>, <&rcc CK_KER_GPU>;
-				gpu-supply =  <&scmi_vddgpu>;
+				power-domains = <&scmi_devpd PD_SCMI_GPU>;
 				feature-domains = <&rifsc STM32MP25_RIFSC_GPU_ID>;
 				status = "disabled";
 			};
